src/components/utils/pagination.ts

```diff
@@ -0,0 +1,71 @@
+import './pagination.css';
+
+interface IPagination {
+  getPageCount: () => number;
+  getPagesArray: () => number[];
+  setPages: (n: number) => number[];
+  pages: (n: number) => number[];
+  pagesElement: (n: number, callback: (e: Event) => void) => HTMLDivElement;
+}
+class Pagination implements IPagination {
+  constructor(private itemsCount: number, private limit: number) {}
+  getPageCount(): number {
+    return Math.ceil(this.itemsCount / this.limit);
+  }
+
+  getPagesArray(): number[] {
+    const totalPages = this.getPageCount();
+    return Array(totalPages)
+      .fill(0)
+      .map((_, i) => i + 1);
+  }
+
+  setPages(num: number): [number, number, number] {
+    const number = num;
+    return [number - 1, number, number + 1];
+  }
```

Comment:
Do we really need 'number' constant or we can immediately return [num - 1, num, num + 1] ?

Reply:
Of course we don't need this, fixed this